File: src/ds-visuals/snippets/tree/delete.py
def delete(self, root, data):
	
	if(root == None):
		print('Error - '+str(data)+' Node Not found')
		return root
	elif(data > root.data):
		root.right = self.delete(root.right, data)
	elif(data < root.data):
		root.left = self.delete(root.left, data)
	else:
		if(root.right == None):


			return root.left
		elif(root.left == None):

			
			return root.right
		else:
			temp = root.right
			while(temp != None and temp.left != None):
				temp = temp.left
			root.data = temp.data
			root.right = self.delete(root.right, temp.data)
	return root

File: src/ds-visuals/snippets/tree/delete.java
TreeNode delete(TreeNode root, int data){

	if(root == null){
		System.out.println("Error - key: "+data+" not found");
		return root;
	}
	else if(data > root.data)
		root.right = delete(root.right, data);
	else if(data < root.data)
		root.left = delete(root.left, data);
	else{
		if(root.right == null){

			
			return root.left;
		}
		else if(root.left == null){
			

			return root.right;
		}
		else{
			TreeNode temp = root.right;
			while(temp != null && temp.left != null)
				temp = temp.left;
			root.data = temp.data;
			root.right = delete(root.right, temp.data);
		}
	}
	return root;
}